### PR TITLE
Custom JDBC column types

### PR DESCRIPTION
**Pull request by marctrem**
This patch allows us to set custom column types.
Please tell me if you want some edits on it.

Thank you,
Marc

### CODE REVIEW

**Comment by JoshRosen** [comment]
Seems reasonable to me; do you mind adding a simple test which exercises this code path?

**Comment by marctrem** [comment]
Sorry for the delay.

I based the tests on the existing "maxlength" feature test.

File: src/main/scala/com/databricks/spark/redshift/RedshiftJDBCWrapper.scala [hunk 1]
```diff
@@ -232,26 +232,32 @@ private[redshift] class JDBCWrapper {
...
-            "TEXT"
-          }
-        case BinaryType => "BLOB"
-        case TimestampType => "TIMESTAMP"
-        case DateType => "DATE"
-        case t: DecimalType => s"DECIMAL(${t.precision},${t.scale})"
-        case _ => throw new IllegalArgumentException(s"Don't know how to save $field to JDBC")
+      val typ: String = if (field.metadata.contains("redshift_type")) {
+        field.metadata.getString("redshift_type")
+      }
```
Comment (JoshRosen) [inline src/main/scala/com/databricks/spark/redshift/RedshiftJDBCWrapper.scala]: Minor style nit: place the `else` on the same line as this brace.

**Comment by JoshRosen** [comment]
LGTM. I'll try to fix the style issue and merge conflicts myself.

**Comment by gatorsmile** [comment]
@JoshRosen We are facing the same issue in the general JDBC data source of Spark. What do you think if we do it in Spark too?

**Comment by JoshRosen** [comment]
@gatorsmile, I wouldn't be opposed.

**Comment by gatorsmile** [comment]
Thanks!

**Comment by gatorsmile** [comment]
@JoshRosen We submitted two solutions:
- https://github.com/apache/spark/pull/16209
- https://github.com/apache/spark/pull/16208

I am wondering which solution you prefer? It sounds like the first one is more user friendly. 
```Scala
val mdb = new MetadataBuilder()
mdb.putString("name", "VARCHAR(128)”)
mdb.putString("comments”, “CLOB(20K)”)
val createTableColTypes = mdb.build().json
df.write.option("createTableColumnTypes", createTableColTypes).jdbc(url, "TEST.DBCOLTYPETEST", properties)
```

**Pull request closed without merging**